front/components/agent_builder/instructions/AgentInstructionsHistory.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuLabel,
+  DropdownMenuRadioGroup,
+  DropdownMenuRadioItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  HistoryIcon,
+  Spinner,
+} from "@dust-tt/sparkle";
+import { useCallback, useMemo } from "react";
+import React from "react";
+
+import { useEditors } from "@app/lib/swr/editors";
+import type {
+  LightAgentConfigurationType,
+  LightWorkspaceType,
+} from "@app/types";
+
+interface AgentInstructionsHistoryProps {
+  history: LightAgentConfigurationType[];
+  selectedConfig: LightAgentConfigurationType | null;
+  onSelect: (config: LightAgentConfigurationType) => void;
+  owner: LightWorkspaceType;
+  agentConfigurationId: string | null;
+}
+
+export function AgentInstructionsHistory({
+  history,
+  onSelect,
+  selectedConfig,
+  owner,
+  agentConfigurationId,
+}: AgentInstructionsHistoryProps) {
+  const { editors, isEditorsLoading } = useEditors({
+    owner,
+    agentConfigurationId,
+    disabled: !agentConfigurationId,
+  });
+
+  const authorMap = useMemo(() => {
+    const map: Record<string, string> = {};
+    editors.forEach((editor) => {
+      map[editor.id] = editor.fullName || editor.firstName;
+    });
+    return map;
+  }, [editors]);
+
+  const formatVersionLabel = useCallback(
+    (config: LightAgentConfigurationType) => {
+      const dateFormatter = new Intl.DateTimeFormat("en-US", {
+        year: "numeric",
+        month: "2-digit",
+        day: "2-digit",
+        hour: "numeric",
+        minute: "numeric",
+        hour12: true,
+      });
+      return config.versionCreatedAt
+        ? dateFormatter.format(new Date(config.versionCreatedAt))
+        : `v${config.version}`;
+    },
+    []
+  );
+
+  const getAuthorName = useCallback(
+    (config: LightAgentConfigurationType) => {
+      if (!config.versionAuthorId) {
+        return "System";
+      }
+      return authorMap[config.versionAuthorId] || "Unknown";
+    },
+    [authorMap]
+  );
+
+  const historyWithPrev = useMemo(() => {
+    const currentVersion = Math.max(...history.map((h) => h.version));
+
+    const sorted = [...history]
+      .filter((config) => config.version !== currentVersion)
+      .sort((a, b) => {
+        const timeA = a.versionCreatedAt
+          ? new Date(a.versionCreatedAt).getTime()
+          : a.version;
+        const timeB = b.versionCreatedAt
+          ? new Date(b.versionCreatedAt).getTime()
+          : b.version;
+
+        if (timeA !== timeB) {
+          return timeB - timeA;
+        }
+        return b.version - a.version;
+      });
```

Comment:
Not clear of the order, you can use `compareAsc` or `compareDesc` to sort it.
```suggestion
      .sort((a, b) =>
        compareAsc(
          a.versionCreatedAt ?? a.version,
          b.versionCreatedAt ?? b.version
        )
      );
```
Feel free to change the correct sort direction.

Reply:
Also out of curiosity, do we have browser version usage? As there is now a nice [`toSorted`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted) that return a new array instead of sorting in place, but might be too new.